MHCollapsibleViewManager.m

```diff
@@ -0,0 +1,228 @@
+    return self;
+}
+
+//called after Initializing Manager
+- (void) initializeData:(NSMutableArray*)dataFromController
+{
+    self.data = [[NSMutableArray alloc] init];
+    for(int i=0; i < [dataFromController count] ; i++)
+    {
+        [data addObject:[MHFilterLabel constructLabel:[dataFromController objectAtIndex:i] checked:false]];
+    }
+    self.numOfRows = (NSInteger)[data count];
+}
+
+//set the cell provided with same settings as corresponding array type
+- (UITableViewCell*)setCellSettings:(UITableViewCell*)cell type:(NSString*)type
+{
+    UITableViewCell *tempCell = nil;
+    int index = 3; //normall cell
+    if([type isEqual: @"ClickedHeader"])
+    {
+        index = 0;
+    }
+    else if([type isEqual: @"NormalHeader"])
+    {
+        index = 1;
+    }
+    else if([type isEqual: @"ClickedCell"])
+    {
+        index = 2;
+    }
+    
+    tempCell = [cellStyles objectAtIndex:index];
+    cell.accessoryType = tempCell.accessoryType;
+    cell.accessoryView = tempCell.accessoryView;
+    cell.textLabel.textColor = tempCell.textLabel.textColor;
+    
+    if(index == 0 || index == 1)
+    {
+        cell.textLabel.text = self.headerTitle;
+        if(self.numOfRows > 0)
+            cell.detailTextLabel.text = [NSString stringWithFormat:NSLocalizedString(@"%i items",), self.numOfRows];
+        else
+            cell.detailTextLabel.text = @"No item";
+    }
+    tempCell = nil;
+    return cell;
+}
+
+//takes in four cells that are formatted or takes nil and creates default cells
+- (void)setCellProperties:(UITableViewCell*)clickedHeader header:(UITableViewCell*)header clickedCell:(UITableViewCell*)clickedCell normalCell:(UITableViewCell*)cell tableView:(UITableView*)tableView
+{
+    //normal cell with no accessory is non selected row
+    if (cell == nil)
+        cell = [self returnAccessorizedCell:UITableViewCellAccessoryNone tableView:tableView];
+    //checkmark shown is the default for when row selected
+    if (clickedCell == nil)
+        clickedCell = [self returnAccessorizedCell:UITableViewCellAccessoryCheckmark tableView:tableView];
+    if (header == nil)
+    {
+        header = [self returnAccessorizedCell:UITableViewCellAccessoryNone tableView:tableView];
+        header.accessoryView = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"downArrow"]];
+    }
+    if (clickedHeader == nil)
+    {
+        clickedHeader = [self returnAccessorizedCell:UITableViewCellAccessoryNone tableView:tableView];
+        //Add up arrow for accessoryView
+        clickedHeader.accessoryView = [[UIImageView alloc] initWithImage:[UIImage imageNamed:@"upArrow"]];
+        //Color is the same as the checkmark for the children rows
+        clickedHeader.textLabel.textColor = [[UIColor alloc] initWithRed:0 green:.48 blue:1.0 alpha:1.0];
+    }
+    
+    //Keeps track of cell properties for the various states
+    self.cellStyles =  [NSArray arrayWithObjects: clickedHeader, header, clickedCell, cell, nil];
```

Comment:
with literals this would be self.cellStyles = @[clickedHeader, header, clickedCell, cell];